### PR TITLE
STOR-989: Remove SC and manifest  file for NFS backend

### PR DESCRIPTION
**Pull request by bertinatto**
Both SC and the manifest file will be created in CI only.

CC @openshift/storage

### CODE REVIEW

File: test/e2e/manifest-nfs.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Test manifest for https://github.com/kubernetes/kubernetes/tree/master/test/e2e/storage/external
 ShortName: azurefile-nfs
 StorageClass:
-  FromName: true
+  FromExistingClassName: azurefile-csi-nfs
```
[the remark this answers is not in the document]
Comment (bertinatto) [reply test/e2e/manifest-nfs.yaml:6]: Done

**Comment by mpatlasov** [comment]
/hold

until this error:

> fail [k8s.io/kubernetes@v1.27.1/test/e2e/storage/external/external.go:285]: getting storage class azurefile-csi-nfs: storageclasses.storage.k8s.io "azurefile-csi-nfs" not found

unresolved (https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_azure-file-csi-driver-operator/58/pull-ci-openshift-azure-file-csi-driver-operator-main-e2e-azure-file-nfs-csi/1672352963101724672)

**Comment by bertinatto** [comment]
> /hold
> 
> until this error:
> 
> > fail [k8s.io/kubernetes@v1.27.1/test/e2e/storage/external/external.go:285]: getting storage class azurefile-csi-nfs: storageclasses.storage.k8s.io "azurefile-csi-nfs" not found
> 
> unresolved (https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_azure-file-csi-driver-operator/58/pull-ci-openshift-azure-file-csi-driver-operator-main-e2e-azure-file-nfs-csi/1672352963101724672)

There's a cyclic dependency with https://github.com/openshift/release/pull/40601, so we'll need to override this job.

**Comment by mpatlasov** [comment]
> > /hold
> > until this error:
> > > fail [k8s.io/kubernetes@v1.27.1/test/e2e/storage/external/external.go:285]: getting storage class azurefile-csi-nfs: storageclasses.storage.k8s.io "azurefile-csi-nfs" not found
> > 
> > 
> > unresolved (https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_azure-file-csi-driver-operator/58/pull-ci-openshift-azure-file-csi-driver-operator-main-e2e-azure-file-nfs-csi/1672352963101724672)
> 
> There's a cyclic dependency with [openshift/release#40601](https://github.com/openshift/release/pull/40601), so we'll need to override this job.

I'm fine with overriding. Feel free to `/unhold` when is ready.

**Comment by bertinatto** [comment]
/hold 
for https://github.com/openshift/release/pull/40601

**Comment by bertinatto** [comment]
https://github.com/openshift/release/pull/40601 has merged, so retest:
/retest
/hold cancel

**Review by tsmetana — APPROVED** [review]
/lgtm

**Pull request merged**